### DISCUSSION TITLE
Have "schedule-only" playlists been overlooked? (Auto-play, no influence on general rotation)

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6114] [Q&A] [answered]
One might have several scheduled playlists, say, genre-based or "show"-based programming, _that should not go into the general rotation_. Currently (without manually editing the LiquidSoap), I can only make such playlists work when setting them to "Enable" and "Playlist Type: General Raotation".

**Have "schedule-only playlists" been overlooked?**

Like probably many of us, I’d fill the day with some nice general rotation mix, but also have "show" or "genre" 2-hour blocks coming from playlists that shouldn’t influence the genereal rotation. Still, the AutoDJ should be able to fill these blocks from either pre-made sequential oder random playlists. **Such, an easy and recognizable simple station programming would be possible.**

A hypothetical example (schedule view):

![Auswahl_199](https://user-images.githubusercontent.com/3706922/221369254-b798cd97-141d-443b-b770-0adc1ec01e73.png)

Using a low weight and a higher weight for general rotation playlists doesn’t really solve the problem.

The playlists _do_ play fine at the specified times, but _influence the general rotation_, i.e. all unscheduled times of the day. Like, I used to have another block "Trance", which is currently unused as "block". Nevertheless, due to _the others_ being played, LiquidSoap "senses" that Trance isn’t played as often as the others and tends to fill all unscheduled times with trance music, while the station _should_ actually play a mixture of Pop and Classic Rock for general rotation.

Show/Genre Playlists would also often be constructed from songs that are also in general rotation, making for an imbalance in song playout frequency if they would appear in general rotation.

**So I really think, apart from the other options, a playlist type of "schedule only" is needed.**

How do other users solve this problem, and how do they think about this proposal?

**Comment by guiltlab** [comment]
see https://github.com/AzuraCast/AzuraCast/issues/5618 I definitely agree scheduled playlists should have the option to be independent from general rotation.

The way I have it on my station is that I have scheduled playlists for everything, giving up the general rotation entirely. Since all time slots are scheduled, general rotation does not happen at all.

**Comment by Moonbase59** [comment]
@guiltlab I favour your (and my) suggestion of having an extra playlist type, **although this _seems_ to have changed in one of the last rolling releases**: My station unexpectedly played "AzuraCast is Live!" for an hour or so shortly after midnight today, until I recognized it didn’t play from the scheduled playlists anymore, so had nothing for "general rotation".

As soon as I duplicated one of my scheduled playlists and removed the scheduled times, it started playing again.

**This is another way of doing it, and it seems to work, _but_ we should at least have some indication, like a "scheduled" tag so we can distinguish the different kinds of playlists more easily.** Using a name scheme is often not feasible, since the scheduled playlists show their name on the schedule view and of course one wants to show "nice names" to the listeners.

Currently on Rolling Release #[768d577](https://github.com/AzuraCast/AzuraCast/commit/768d5772de44887e4345925bdf9206df289397e4) (2023-02-25 13:40).

Still have to test the "requests from this playlists only during the scheduled times" feature. Would be a nice thing to avoid having users request songs from a "wrong" genre.

Using a "weight" makes still sense, so I can schedule two playlists for the same time(s), possibly ending up with a weighted mixture of songs (and requests) during that "block" (see "Krautrock/Prog Rock" at 20:00–22:00 on Thursdays above). Which actually worked last time—both having a weight of "3" made for playing one song from the "Krautrock" playlist, one song from "Prog Rock" alternately.

**Comment by Moonbase59** [comment, marked as answer]
Functionally, this works okay in the latest rolling releases (I’m currently on Rolling Release #[5cbf9c9](https://github.com/AzuraCast/AzuraCast/commit/5cbf9c91cae1a234c428cf43d769fe2349e05bd9) (2023-02-28 8:15)), so for discussion’s sake, I consider this "answered".

It’d still be nice to distinguish scheduled playlists more easily (I currently have 39 playlists to do some genre-based programming during the days of the week), but as I read, [tagging these is planned](https://features.azuracast.com/suggestions/364871/add-scheduled-tag-to-playlist-view-for-easier-identification-of-scheduled-playli).

Regarding playlist song requests, I’ll open [another discussion](https://github.com/AzuraCast/AzuraCast/discussions/6125).
